### ISSUE TITLE
infer.py: error: unrecognized arguments: fairseqlm

### ISSUE THREAD

**Issue by anujsaraswat741** [pytorch/fairseq#2478]
Trace

python examples/speech_recognition/infer.py /home/anuj/temp/LibriSpeech/dev-other/ --task audio_pretraining --nbest 1 --path /home/anuj/pretrained_models/wav2vec2_vox_960h.pt --gen-subset dev_other --results-path /home/anuj/fairseq/--w2l-decoder fairseqlm --lm-model /home/anuj/language_models/lm_librispeech_word_transformer.pt --lm-weight 2 --word-score 1 --sil-weight 0 --criterion ctc --labels ltr --max-tokens 40000 --post-process letter

usage: infer.py [-h] [--no-progress-bar] [--log-interval N]
                [--log-format {json,none,simple,tqdm}]
                [--tensorboard-logdir DIR] [--seed N] [--cpu] [--tpu] [--bf16]
                [--fp16] [--memory-efficient-bf16] [--memory-efficient-fp16]
                [--fp16-no-flatten-grads] [--fp16-init-scale FP16_INIT_SCALE]
                [--fp16-scale-window FP16_SCALE_WINDOW]
                [--fp16-scale-tolerance FP16_SCALE_TOLERANCE]
                [--min-loss-scale D]
                [--threshold-loss-scale THRESHOLD_LOSS_SCALE]
                [--user-dir USER_DIR] [--empty-cache-freq EMPTY_CACHE_FREQ]
                [--all-gather-list-size ALL_GATHER_LIST_SIZE]
                [--model-parallel-size N]
                [--checkpoint-suffix CHECKPOINT_SUFFIX]
                [--quantization-config-path QUANTIZATION_CONFIG_PATH]
                [--profile]
                [--criterion {legacy_masked_lm_loss,sentence_prediction,label_smoothed_cross_entropy,label_smoothed_cross_entropy_with_alignment,ctc,wav2vec,nat_loss,composite_loss,masked_lm,sentence_ranking,adaptive_loss,cross_entropy,vocab_parallel_cross_entropy}]
                [--scoring {sacrebleu,bleu,wer}]
                [--tokenizer {nltk,moses,space}]
                [--bpe {bytes,hf_byte_bpe,bert,subword_nmt,byte_bpe,characters,gpt2,sentencepiece,fastbpe}]
                [--optimizer {adadelta,adafactor,adam,adagrad,sgd,nag,adamax,lamb}]
                [--lr-scheduler {tri_stage,polynomial_decay,triangular,reduce_lr_on_plateau,fixed,cosine,inverse_sqrt}]
                [--task TASK] [--num-workers N]
                [--skip-invalid-size-inputs-valid-test] [--max-tokens N]
                [--max-sentences N] [--required-batch-size-multiple N]
                [--dataset-impl FORMAT] [--data-buffer-size N]
                [--gen-subset SPLIT] [--num-shards N] [--shard-id ID]
                [--distributed-world-size N]
                [--distributed-rank DISTRIBUTED_RANK]
                [--distributed-backend DISTRIBUTED_BACKEND]
                [--distributed-init-method DISTRIBUTED_INIT_METHOD]
                [--distributed-port DISTRIBUTED_PORT] [--device-id DEVICE_ID]
                [--distributed-no-spawn] [--ddp-backend {c10d,no_c10d}]
                [--bucket-cap-mb MB] [--fix-batches-to-gpus]
                [--find-unused-parameters] [--fast-stat-sync]
                [--broadcast-buffers] [--distributed-wrapper {DDP,SlowMo}]
                [--slowmo-momentum SLOWMO_MOMENTUM]
                [--slowmo-algorithm {LocalSGD,SGP}]
                [--localsgd-frequency LOCALSGD_FREQUENCY]
                [--nprocs-per-node N] [--path FILE]
                [--remove-bpe [REMOVE_BPE]] [--quiet] [--model-overrides DICT]
                [--results-path RESDIR] [--beam N] [--nbest N] [--max-len-a N]
                [--max-len-b N] [--min-len N] [--match-source-len]
                [--no-early-stop] [--unnormalized] [--no-beamable-mm]
                [--lenpen LENPEN] [--unkpen UNKPEN]
                [--replace-unk [REPLACE_UNK]] [--sacrebleu]
                [--score-reference] [--prefix-size PS]
                [--no-repeat-ngram-size N] [--sampling] [--sampling-topk PS]
                [--sampling-topp PS] [--temperature N]
                [--diverse-beam-groups N] [--diverse-beam-strength N]
                [--diversity-rate N] [--print-alignment] [--print-step]
                [--iter-decode-eos-penalty N] [--iter-decode-max-iter N]
                [--iter-decode-force-max-iter] [--iter-decode-with-beam N]
                [--iter-decode-with-external-reranker] [--retain-iter-history]
                [--retain-dropout]
                [--retain-dropout-modules RETAIN_DROPOUT_MODULES [RETAIN_DROPOUT_MODULES ...]]
                [--decoding-format {unigram,ensemble,vote,dp,bs}]
                [--kspmodel KSPMODEL] [--wfstlm WFSTLM]
                [--rnnt_decoding_type RNNT_DECODING_TYPE]
                [--lm-weight LM_WEIGHT] [--rnnt_len_penalty RNNT_LEN_PENALTY]
                [--w2l-decoder {viterbi,kenlm,fairseqlm}] [--lexicon LEXICON]
                [--unit-lm] [--kenlm-model KENLM_MODEL]
                [--beam-threshold BEAM_THRESHOLD]
                [--beam-size-token BEAM_SIZE_TOKEN] [--word-score WORD_SCORE]
                [--unk-weight UNK_WEIGHT] [--sil-weight SIL_WEIGHT]
                [--dump-emissions DUMP_EMISSIONS]
                [--dump-features DUMP_FEATURES]
                [--load-emissions LOAD_EMISSIONS] [--zero-infinity]
                [--wer-args WER_ARGS] [--momentum M] [--weight-decay WD]
                [--force-anneal N] [--lr-shrink LS] [--warmup-updates N]
                [--sample-rate SAMPLE_RATE] [--normalize]
                [--max-sample-size MAX_SAMPLE_SIZE]
                [--min-sample-size MIN_SAMPLE_SIZE] [--enable-padding]
                [--no-min-cropping] [--labels LABELS]
                data
infer.py: error: unrecognized arguments: fairseqlm

**Comment by kamakshi-malhotra** [comment]
Hi @anujsaraswat741 , I recieved same error. How were you able to resolve this?
